### ISSUE TITLE
Error bars in bar plot are not displayed

### ISSUE THREAD

**Issue by btlorch** [matlab2tikz/matlab2tikz#812]
I needed to make a bar plot with error bars and found a nice function called barwitherr on [fileexchange](http://www.mathworks.com/matlabcentral/fileexchange/30639-barwitherr-errors-varargin-). However the error bars are not exported to tikz. I think I found the responsible lines and a manual workaround, but couldn't come up with a complete solution.

In line 134 of the barwitherr.m file, the markers are disabled.
`set(hErrorbar(col), 'marker', 'none')`

Then around line 1400 of matlab2tikz.m, there is a check whether there is anything to plot. As the markers have been disabled and the lines that would connect the error bars have been disabled, the if-condition evaluates to true and the function decides that there is nothing to plot. As a consequence, the error bars are not exported.
```
hasLines   = ~isNone(lineStyle) && lineWidth > 0;
hasMarkers = ~isNone(marker);
if ~hasLines && ~hasMarkers
    return
end
```

As a quick workaround one can set the marker property in the barwitherr function instead of 'none' to '.'. Then matlabt2tikz will write out something like
`\addplot [color=black,only marks,mark=*,mark options={solid},forget plot]`
which gives some fat filled circles as markers. To get rid of the markers, one can set its size to a very small value, e.g.
`\addplot [color=black,only marks,mark=*,mark options={mark size=.25},forget plot]`

Hope this can be resolved easily or someone can make use of the workaround!

**Comment by egeerardyn** [comment]
Thank you for bringing this to our attention.
However, I do wonder what your figure looks like when the error bars don't show lines and no markers. 

Could you also show a screenshot of the figure (and maybe even some code to reproduce that plot)? That would help us along when we analyze this further and see whether we can make sure that matlab2tikz reproduces your plot better.

**Comment by btlorch** [comment]
Sure, here is some arbitrary example:
(Make sure that you have barwitherr from fileexchange in your path)

```
y = [8, 7, 9; 7, 4, 5; 6, 8, 8];
err = [1, 2, 1; 1, 2, 2; 1, 0, 2];

figure;
[hbar, hErrorbar] = barwitherr(err, y);
ylim([1.5 10.5]);
matlab2tikz('test.tikz', 'height', '\fheight', 'width', '\fwidth');
```

![test](https://cloud.githubusercontent.com/assets/11959715/11066568/b3975e0e-87c0-11e5-8abd-eb9c34ad7c3c.png)

Sorry for being unclear. The [errorbar](http://uk.mathworks.com/help/matlab/ref/errorbar.html) function that is provided by Matlab typically draws connecting lines between the mean values of the individual error bars, which I want to avoid obviously.

![errorbar from Matlab website](http://uk.mathworks.com/help/releases/R2015b/examples/graphics/SymmetricErrorBarsExample_01.png)

**Comment by egeerardyn** [comment]
I am very confident we can fix this on our end.

If you want to patch your copy of `m2t` in the meanwhile, you can probably change
```matlab
if ~hasLines && ~hasMarkers
```
to
```matlab
if ~hasLines && ~hasMarkers && nargin < 2
```
to fix this.

I will patch the official version in a cleaner way.

**Comment by egeerardyn** [comment]
Should be fixed by #813.
